graphics/color_quantizer.cpp

```diff
@@ -0,0 +1,268 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common/stack.h"
+
+#include "graphics/color_quantizer.h"
+#include "graphics/palette.h"
+
+namespace Graphics {
+
+// This code is heavily based on "Color Quantization using Octrees" by Dean
+// Clark, published in - I think - the January 1996 issue of Dr. Dobb's Journal.
+//
+// FIXME: The intention is that this should be moved to common code, but we keep
```

Comment:
Is this FIXME fully addressed by this PR?

Reply:
Yes, that's a cut and past error. Will fix. Thanks!